### PR TITLE
feat: module context over gRPC (protobuf changes)

### PR DESCRIPTION
**Pull request by matt2e**
Smaller PR for this: https://github.com/TBD54566975/ftl/pull/1311
Just includes protobuf changes

### CODE REVIEW

File: backend/protos/xyz/block/ftl/v1/ftl.proto [hunk 1]
```diff
@@ -16,6 +16,41 @@ message PingResponse {
   optional string not_ready = 1;
 }
 
+message ModuleContextRequest {
+  string module = 1;
+}
+
+message ModuleContextResponse {
+  message Ref {
+    optional string module = 1;
```
Comment (alecthomas) [inline backend/protos/xyz/block/ftl/v1/ftl.proto:25]: I think we probably don't even need this in hindsight. The module isn't going to care where the secret comes from, just the name of it...?

File: backend/protos/xyz/block/ftl/v1/ftl.proto [hunk 2]
```diff
@@ -16,6 +16,41 @@ message PingResponse {
...
+      POSTGRES = 0;
+  }
+
+  message DSN {
+    string name = 1;
+    DBType type = 2;
+    string dsn = 3;
+  }
+
+  repeated Config configs = 1;
```
Comment (alecthomas) [inline backend/protos/xyz/block/ftl/v1/ftl.proto:49]: Given the comment above we could potentially just use `map<string, bytes>`

**Review by alecthomas — APPROVED** [review]

**Review by worstell — APPROVED** [review]

File: backend/protos/xyz/block/ftl/v1/ftl.proto (hunk 1 above)
Comment (matt2e) [reply backend/protos/xyz/block/ftl/v1/ftl.proto:25]: Can't there be naming conflicts though?
Module config named X, vs global config named X

Comment (alecthomas) [reply backend/protos/xyz/block/ftl/v1/ftl.proto:25]: Not from a module's perspective - all it sees is a flat set of keys

Comment (matt2e) [reply backend/protos/xyz/block/ftl/v1/ftl.proto:25]: Ah, I didn't realise that. that'll simplify some still in the main PR as well

**Pull request merged**